stdlib/std.jsonnet

```diff
@@ -35,18 +35,6 @@ limitations under the License.
   toString(a)::
     if std.type(a) == 'string' then a else '' + a,
 
-  substr(str, from, len)::
```

Comment:
If you leave these in, it won't break the Go implementation.  They should be overridden by the C++ ones in desugarer.cpp

Reply:
(All you have to do is revert the changes in this file.)

Reply:
Ok the problem there is that this code (in desugarer.cpp) is adding them even though they're already in std_obj->fields.

```
        // Bind 'std' builtins that are implemented natively.
        DesugaredObject::Fields &fields = std_obj->fields;
        for (unsigned long c = 0; c <= max_builtin; ++c) {
            const auto &decl = jsonnet_builtin_decl(c);
            Identifiers params;
            for (const auto &p : decl.params)
                params.push_back(id(p));
            fields.emplace_back(ObjectField::HIDDEN,
                                str(decl.name),
                                make<BuiltinFunction>(E, encode_utf8(decl.name), params));
        } 
        fields.emplace_back(
            ObjectField::HIDDEN, str(U"thisFile"), str(decode_utf8(ast->location.file)));
```

What's necessary is to build a DesugaredObject::Fields with only the fields we want.

Something like this (I haven't tried compiling it):

```
        std::unordered_map<Identifier, DesugaredObject::Field> fields;
        for (const auto &jsonnet_field : std_obj->fields) {
            fields[jsonned_field.name] = jsonnet_field;
        }
        for (unsigned long c = 0; c <= max_builtin; ++c) {
            const auto &decl = jsonnet_builtin_decl(c);
            Identifiers params;
            for (const auto &p : decl.params)
                params.push_back(id(p));
            fields[decl.name] = DesugaredObject::Field(
                ObjectField::HIDDEN, str(decl.name),
                make<BuiltinFunction>(E, encode_utf8(decl.name), params));
        } 
        fields[decl.name] = DesugaredObject::Field(
            ObjectField::HIDDEN, str(U"thisFile"), str(decode_utf8(ast->location.file)));
        std_obj->fields.clear();
        for (const auto &item : fields) {
            std_obj->fields.push_back(item.second);
        }
```

Reply:
I've implemented something along these lines (though the details are different).  PTAL.